third_party/move/move-compiler-v2/tests/checking/receiver/generic_calls.move

```diff
@@ -0,0 +1,44 @@
+module 0x42::m {
+
+    struct S<T> { x: T }
+
+    // Call styles
+
+    fun receiver<T>(self: S<T>, y: T) {
+        self.x = y;
+    }
+
+    fun receiver_ref<T>(self: &S<T>, _y: T) {
+    }
+
+    fun receiver_ref_mut<T>(self: &mut S<T>, y: T) {
+        self.x = y
+    }
+
+    fun receiver_more_generics<T, R>(self: S<T>, _y: R) {
+    }
+
+    fun test_call_styles(s: S<u64>, x: u64) {
+        s.receiver(x);
+        s.receiver_ref(x);
+        s.receiver_ref_mut(x);
+        s.receiver_more_generics(22)
```

Comment:
1. Could we please add a "passing" case for the turbofish operator (`::<>`)?
2. Could we please add a case with `inline` receiver functions?